src/pdm/cli/commands/run.py

```diff
@@ -216,12 +228,24 @@ def _run_task(
         kind, _, value, options = task
         shell = False
         if kind == "cmd":
-            if not isinstance(value, list):
-                value = shlex.split(str(value))
-            args = value + list(args)
+            interpolations = 0
+            if isinstance(value, str):
+                cmd, count = interpolate(value, args)
+                interpolations += count
+                value = shlex.split(cmd)
+            else:
+                agg = [interpolate(part, args) for part in value]
+                interpolations += sum(row[1] for row in agg)
+                parts = (
+                    shlex.split(interpolated) if count else [interpolated]
+                    for interpolated, count in agg
+                )
+                value = list(itertools.chain(*parts))
```

Comment:
I can read this, but we may need some minimum comments.

Reply:
I added both comments to explain and typing for the intermediate `parts` variable